src/ghdcbot/config/loader.py

```
@@ -16,19 +16,48 @@
 _ENV_PATTERN = re.compile(r"\$\{([A-Z0-9_]+)\}")
 
 
+class _ConfigLoader(yaml.SafeLoader):
+    """YAML loader with !include relative to the config file directory."""
+
+    def __init__(self, stream, *, config_dir: Path) -> None:
+        super().__init__(stream)
+        self.config_dir = config_dir
+
+
+def _construct_include(loader: _ConfigLoader, node: yaml.Node) -> Any:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_construct_include`

(ANN401)